src/app/Mappings/components/AddEditMappingModal.tsx

```diff
@@ -60,6 +60,8 @@ const AddEditMappingModal: React.FunctionComponent<IAddEditMappingModalProps> =
     mappingType
   );
 
+  const isEveryItemFilled = () => builderItems.every((item) => item.source && item.target);
```

Comment:
I don't see what we gain by making this a function, since it is initialized and called all in the same function scope as `builderItems`, and doesn't take any arguments. If we put it after the `useState` below it could just be a variable:

```ts
const isEveryItemFilled = builderItems.every((item) => item.source && item.target);
```

and then similarly the prop type for passing it to MappingBuilder would become a boolean.

I think making things like this a function is only useful if they need to be used in multiple scopes with arguments from each scope, like if we had two different variants of this modal that both had their own set of `builderItem`s.

Reply:
Yes functions are best for reusability but sometimes just for readability.
That said it's called in two different places.